app/src/main/kotlin/com/wire/android/ui/home/conversations/DownloadedAssetDialog.kt

```diff
@@ -0,0 +1,100 @@
+package com.wire.android.ui.home.conversations
+
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Environment
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import com.wire.android.R
+import com.wire.android.appLogger
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.util.copyDataToTempFile
+import com.wire.android.util.getMimeType
+import com.wire.android.util.permission.WriteToExternalStorageRequestFlow
+import com.wire.android.util.saveFileDataToDownloadsFolder
+import java.io.File
+
+@Composable
+fun DownloadedAssetDialog(conversationViewModel: ConversationViewModel) {
```

Comment:
DownloadedAssetDialog is going to be really hard coupled on ConversationViewModel() is it something we want ?

Reply:
i think it would make more sense to expose some lambda's to the consumer so that we do not make it coupled on any viewmodel

Reply:
I just followed what had been previously implemented by the `DeleteMessageDialog`. But you are right, we should try to decouple this and maybe substitute it for some lambdas. Good catch!